flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerServices.java

```diff
@@ -490,14 +497,17 @@ private static void checkTempDirs(String[] tmpDirs) throws IOException {
 	}
 
 	public static ResourceProfile computeSlotResourceProfile(int numOfSlots, long managedMemorySize) {
-		int managedMemoryPerSlotMB = (int) bytesToMegabytes(managedMemorySize / numOfSlots);
+		// TODO: before operators separate on-heap/off-heap managed memory, we use on-heap managed memory to denote total managed memory
+		return computeSlotResourceProfile(numOfSlots, Collections.singletonMap(MemoryType.HEAP, managedMemorySize));
```

Comment:
I'm wondering if we shouldn't in-line this call into the ResourceManager, with a dedicated comment as to why setting only the HEAP memory type isn't a problem. It's a bit weird to have a method in this class if it never even called in the context of a task executor.

Reply:
True, we could but it should completely disappear soon after #10161.